**Windows** [Build 4131](http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=85562&buildTypeId=WF_PullRequest_Windows) outcome was **FAILURE** using a merge of d9600b02a37671e8a511fbdb5a7920baf47e9a87
Summary: Tests failed: 1 (1 new), passed: 3314, ignored: 190 Build time: 01:10:27
### Failed tests
```
org.jboss.as.test.xts.wsat.client.ATTestCase.testWSATApplicationException: 	at com.sun.proxy.$Proxy25.runTestMethod(Unknown Source)
Caused by: java.io.IOException: Unable to invoke invoke(), status=WAITING
	at org.jboss.remotingjmx.protocol.v2.ClientConnection$TheConnection.invoke(ClientConnection.java:1072)
	at org.jboss.as.arquillian.container.ManagementClient$MBeanConnectionProxy.invoke(ManagementClient.java:560)
	at javax.management.MBeanServerInvocationHandler.invoke(MBeanServerInvocationHandler.java:292)
	at com.sun.proxy.$Proxy25.runTestMethod(Unknown Source)
------- Stdout: -------
17:47:38,167 INFO  [org.jboss.as.test.xts.wsat.client.ATClient] (pool-3-thread-1) wsdlLocation for service: http://[::1]:8080/wsat-test/ATService1?wsdl
17:47:38,175 INFO  [org.apache.cxf.wsdl.service.factory.ReflectionServiceFactoryBean] (pool-3-thread-1) Creating Service {http://www.jboss.com/jbossas/test/xts/wsat/at/}ATService1 from WSDL: http://[::1]:8080/wsat-test/ATService1?wsdl
17:47:38,184 INFO  [org.jboss.as.test.xts.wsat.client.ATClient] (pool-3-thread-1) wsdlLocation for service: http://[::1]:8080/wsat-test/ATService2?wsdl
17:47:38,193 INFO  [org.apache.cxf.wsdl.service.factory.ReflectionServiceFactoryBean] (pool-3-thread-1) Creating Service {http://www.jboss.com/jbossas/test/xts/wsat/at/}ATService2 from WSDL: http://[::1]:8080/wsat-test/ATService2?wsdl
17:47:38,200 INFO  [org.jboss.as.test.xts.wsat.client.ATClient] (pool-3-thread-1) wsdlLocation for service: http://[::1]:8080/wsat-test/ATService3?wsdl
17:47:38,209 INFO  [org.apache.cxf.wsdl.service.factory.ReflectionServiceFactoryBean] (pool-3-thread-1) Creating Service {http://www.jboss.com/jbossas/test/xts/wsat/at/}ATService3 from WSDL: http://[::1]:8080/wsat-test/ATService3?wsdl
17:47:38,224 INFO  [org.apache.cxf.wsdl.service.factory.ReflectionServiceFactoryBean] (pool-3-thread-1) Creating Service {http://docs.oasis-open.org/ws-tx/wscoor/2006/06}ActivationService from WSDL: jar:file:/C:/BuildAgent/work/a31d203e70e89f90/dist/target/wildfly-10.0.0.Final-SNAPSHOT/modules/system/layers/base/org/jboss/xts/main/jbossxts-5.2.12.Final.jar!/org/oasis_open/docs/ws_tx/wscoor/_2006/_06/wsdl/wscoor-activation-binding.wsdl
17:47:38,228 INFO  [org.apache.cxf.wsdl.service.factory.ReflectionServiceFactoryBean] (pool-3-thread-1) Creating Service {http://docs.oasis-open.org/ws-tx/wscoor/2006/06}ActivationService from WSDL: jar:file:/C:/BuildAgent/work/a31d203e70e89f90/dist/target/wildfly-10.0.0.Final-SNAPSHOT/modules/system/layers/base/org/jboss/xts/main/jbossxts-5.2.12.Final.jar!/org/oasis_open/docs/ws_tx/wscoor/_2006/_06/wsdl/wscoor-activation-binding.wsdl
17:47:38,252 INFO  [org.apache.cxf.wsdl.service.factory.ReflectionServiceFactoryBean] (pool-3-thread-1) Creating Service {http://docs.oasis-open.org/ws-tx/wscoor/2006/06}RegistrationService from WSDL: jar:file:/C:/BuildAgent/work/a31d203e70e89f90/dist/target/wildfly-10.0.0.Final-SNAPSHOT/modules/system/layers/base/org/jboss/xts/main/jbossxts-5.2.12.Final.jar!/org/oasis_open/docs/ws_tx/wscoor/_2006/_06/wsdl/wscoor-registration-binding.wsdl
	at com.sun.proxy.$Proxy25.runTestMethod(Unknown Source)
Caused by: java.io.IOException: Unable to invoke invoke(), status=WAITING
	at org.jboss.remotingjmx.protocol.v2.ClientConnection$TheConnection.invoke(ClientConnection.java:1072)
	at org.jboss.as.arquillian.container.ManagementClient$MBeanConnectionProxy.invoke(ManagementClient.java:560)
	at javax.management.MBeanServerInvocationHandler.invoke(MBeanServerInvocationHandler.java:292)
	... 86 more


```

--- a/ejb3/src/main/java/org/jboss/as/ejb3/remote/protocol/versionone/VersionOneProtocolChannelReceiver.java
+++ b/ejb3/src/main/java/org/jboss/as/ejb3/remote/protocol/versionone/VersionOneProtocolChannelReceiver.java
@@ -310,8 +310,10 @@ public class VersionOneProtocolChannelReceiver implements Channel.Receiver, Depl
             if (EjbLogger.REMOTE_LOGGER.isDebugEnabled()) {
                 EjbLogger.REMOTE_LOGGER.debugf("Received cluster removal notification for cluster %s", registry.getGroup());
             }
-            // when the membership of the cluster being left is 1, we are the last node
-            if (registry.getEntries().keySet().size() == 1) {
+            Map.Entry<String, List<ClientMapping>> localEntry = registry.getEntry(registry.getGroup().getLocalNode());
+            Map<String, List<ClientMapping>> entries = registry.getEntries();
+            // Send cluster removed message only if we are the only remaining member of the cluster that we are leaving
+            if ((localEntry != null) ? (entries.size() == 1) && entries.containsKey(localEntry.getKey()) : entries.isEmpty()) {
                 this.sendClusterRemovedMessage(registry);
             }
         } catch (IOException ioe) {

--- a/testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/remote/RemoteFailoverTestCase.java
+++ b/testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/remote/RemoteFailoverTestCase.java
@@ -68,7 +68,6 @@ import org.junit.runner.RunWith;
  */
 @RunWith(Arquillian.class)
 @RunAsClient
-@Ignore("WFLY-3532")
 public class RemoteFailoverTestCase extends ClusterAbstractTestCase {
     private static final Logger log = Logger.getLogger(RemoteFailoverTestCase.class);
     private static final String MODULE_NAME = "remote-failover-test";
@@ -280,9 +279,9 @@ public class RemoteFailoverTestCase extends ClusterAbstractTestCase {
         }
     }
 
-    // @Ignore("re-enable when WFLY-3532 resoolved")
     @InSequence(3)
     @Test
+    @Ignore("WFLY-3532")
     public void testConcurrentFailover() throws Exception {
         ContextSelector<EJBClientContext> selector = EJBClientContextSelector.setup(CLIENT_PROPERTIES);
         try (EJBDirectory directory = new RemoteEJBDirectory(MODULE_NAME)) {

